sample_app/src/test/java/com/discord/sampleapp/OverlappingPanelsIntegrationTest.kt

```diff
@@ -0,0 +1,309 @@
+package com.discord.sampleapp
+
+import android.app.Activity
+import android.os.Build
+import android.os.Looper
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.View
+import com.discord.panels.OverlappingPanelsLayout
+import com.discord.panels.PanelState
+import com.google.common.truth.Truth
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows
+import org.robolectric.android.controller.ActivityController
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+@Config(sdk = [Build.VERSION_CODES.P])
+@LooperMode(LooperMode.Mode.PAUSED)
+@RunWith(RobolectricTestRunner::class)
+class OverlappingPanelsIntegrationTest {
+
+  private lateinit var activityController: ActivityController<Activity>
+  private lateinit var activity: Activity
+
+  private lateinit var overlappingPanelsLayout: OverlappingPanelsLayout
+
+  @Before
+  fun setUp() {
+    activityController = Robolectric.buildActivity(Activity::class.java)
+    activity = activityController.get()
+
+    val root = LayoutInflater.from(activity)
+      .inflate(
+        R.layout.main_activity /* resource */,
+        null /* root */
+      )
+
+    overlappingPanelsLayout = root.findViewById(R.id.overlapping_panels) as OverlappingPanelsLayout
+
+    // OverlappingPanelsLayout initializes its references to the child panels in onLayout, so
```

Comment:
As requested by @lytefast (in Discord chat), I've added a code comment to explain why we need to explicitly call `layout` here.